src/frontend/e2e/pages/filter.ts

```diff
@@ -99,6 +99,15 @@ export abstract class FilterSample {
     }
     return filterDate;
   }
+
+  //calculate days between two different dates, takings today as initial date to measure
+  public static getDaysFromDateRange(filteredDate: string){
```

Comment:
Why do we need this?

Reply:
i need to create a random past date and used in the request

Reply:
> 

We already have function for this, have a look utils/common.ts

Reply:
> i need to create a random past date and used in the request

Having reviewed this more thoroughly I don't agree with this approach.
1. Sampleutil.getSampleresponseData() already gives you date in the past. if it doesn't then that is a bug we need to fix there, without creating another function
2. Sampleutil.getSampleresponseData() optioooonally takes defaults and if we want to return specific value for any of the attributes we need to feed as input instead of generating and then overwriting.
3. If we really have, then go that way, it make more sense to refactor convertDaysToDate function (rename, change parameters and add logic to handle string date)

Reply:
i have removed this function